src/Infrastructure.VS/VsVersion/VsVersionProvider.cs

```diff
@@ -37,22 +37,23 @@ internal class VsVersionProvider : IVsVersionProvider
 
         [ImportingConstructor]
         public VsVersionProvider([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, ILogger logger)
-            : this(serviceProvider, new SetupConfiguration(), logger)
+            : this(serviceProvider, new SetupConfigurationProvider(), logger)
         {
         }
 
-        internal VsVersionProvider(IServiceProvider serviceProvider, ISetupConfiguration setupConfiguration, ILogger logger)
+        internal VsVersionProvider(IServiceProvider serviceProvider, ISetupConfigurationProvider setupConfigurationProvider, ILogger logger)
```

Comment:
I would pass Func<ISetupConfiguration> but I got an error in the tests `ype 'Func<ISetupConfiguration>' from assembly 'SonarLint.VisualStudio.Infrastructure.VS, Version=4.34.0.0, Culture=neutral, PublicKeyToken=null' cannot be used across assembly boundaries because it has a generic type argument that is an embedded interop type.`, and since we reference the setup config dll as a nuget package, I couldn't set EmbedInteropTypes=False easily on it.

Reply:
Fair enough. Don't forget about `delegate`s. That seems to compile and run ok.
Again, not sure it's worth changing.